cli/src/ui/commands/Chat.tsx

```diff
@@ -78,7 +91,94 @@ const CliChat: FC<CliChatProps> = ({ sId: requestedSId }) => {
     setIsSelectingNewAgent(true);
   }, []);
 
-  const commands = createCommands({ triggerAgentSwitch });
+  const clearFiles = useCallback(() => {
+    setUploadedFiles([]);
+    setPendingFiles([]);
+    setIsUploadingFiles(false);
+  }, []);
+
+  const showAttachDialog = useCallback(async () => {
+    await clearTerminal();
+    setShowFileSelector(true);
+  }, []);
+
+  // Helper to create a conversation for file uploads if none exists
+  // Only useful for uploading files to the first message
+  const createConversationForFiles = useCallback(async () => {
+    if (!selectedAgent || !me || meError || isMeLoading) {
+      return null;
+    }
+
+    const dustClient = await getDustClient();
+    if (!dustClient) {
+      setError("Authentication required. Run `dust login` first.");
+      return null;
+    }
+
+    const convRes = await dustClient.createConversation({
+      title: "File Upload",
+      visibility: "unlisted",
+      contentFragments: [],
+    });
+
+    if (convRes.isErr()) {
+      setError(`Failed to create conversation: ${convRes.error.message}`);
+      return null;
+    }
+
+    setConversationId(convRes.value.conversation.sId);
+    return convRes.value.conversation.sId;
+  }, [selectedAgent, me, meError, isMeLoading]);
+
+  const handleFileSelected = useCallback(
+    async (filePathOrPaths: string | string[]) => {
+      setShowFileSelector(false);
+      try {
+        if (Array.isArray(filePathOrPaths)) {
+          const fileInfos = await Promise.all(
+            filePathOrPaths.map((p) => validateAndGetFileInfo(p))
+          );
+          if (!conversationId) {
+            const newConvId = await createConversationForFiles();
+            if (!newConvId) {
+              return;
+            } // error already handled
+            setPendingFiles(fileInfos);
+            setIsUploadingFiles(true);
+          } else {
+            setPendingFiles(fileInfos);
+            setIsUploadingFiles(true);
+          }
+        } else {
```

Comment:
can we try to reduce the duplication and branching here ? Overall these nested if/else are a bit hard to reason about